src/BloomBrowserUI/bookEdit/js/editViewFrame.js

```diff
@@ -137,10 +110,154 @@ function initializeReaderSetupDialog() {
     $('#synphonyConfig').dialog("close");
+}
+// "endregion" ReaderSetup dialog
+
+// "region" Add Page dialog
+function FindOrCreateAddPageDiv(templatesJSON, descriptionLabel, blankPreviewMsg) {
+
+    var dialogContents = $('<div id="addPageConfig"/>').appendTo($('body'));
+
+    var html = "<iframe id=\"addPage_frame\" src=\"/bloom/pageChooser/page-chooser-main.htm\" scrolling=\"no\" style=\"width: 100%; height: 100%; border-width: 0; margin: 0\"></iframe>";
+
+    dialogContents.append(html);
+
+    // When the page chooser loads, get the iframe holding it to resize to what's inside
+    $('#addPage_frame').load(function() {
+        localizeDialogContents(dialogContents, descriptionLabel, blankPreviewMsg);
+        initializeAddPageDialog(templatesJSON);
+    });
+
+    return dialogContents;
+}
+
+//noinspection JSUnusedGlobalSymbols
+// method called from EditingModel.cs
+function showAddPageDialog(templatesJSON) {
+
+    var theDialog;
+    var parentElement = document.getElementById('page').contentWindow;
+    parentElement.localizationManager.loadStrings(getAddPageDialogLocalizedStrings(), null, function() {
+
+        var title = parentElement.localizationManager.getText('AddPageDialog.Title', 'Add Page...');
+        var addButtonText = parentElement.localizationManager.getText('AddPageDialog.AddPageButton', 'Add This Page');
+        var descriptionLabel = parentElement.localizationManager.getText('AddPageDialog.DescriptionLabel', 'Description');
+        var blankPreviewMsg = parentElement.localizationManager.getText('AddPageDialog.PreviewMessage',
+            'This will contain a preview of a template page when one is selected.');
+        var dialogContents = FindOrCreateAddPageDiv(templatesJSON, descriptionLabel, blankPreviewMsg);
+
+        // Set dialog size (borrowed from reader tools setup dialog; how much of this is useful?!)
+        var h = 750;
+        var w = 940;
+
+        // This height and width will fit inside the "800 x 600" settings
+        var sw = document.body.scrollWidth;
+        if (sw < 583) {
+            h = 460;
+            w = 390;
+        }
+
+        // This height and width will fit inside the "1024 x 586 Low-end netbook with windows Task bar" settings
+        else if ((sw < 723) || (window.innerHeight < 583)) {
+            h = 460;
+            w = 580;
+        }
+
+        theDialog = $(dialogContents).dialog({
+            autoOpen: false,
+            modal: true,
+            width: w,
+            height: h,
+            position: {
+                my: "left top", at: "left top", of: window
+            },
+            title: title,
+            buttons: {
+                OK: {
+                    text: addButtonText,
+                    icons: {
+                        primary: "ui-icon-plusthick"
+                    },
+                    click: function () {
+                        document.getElementById('addPage_frame').contentWindow.postMessage('AddSelectedPage', '*');
```

Comment:
Learn something new every day! the asterisk is the 'context' parameter. I think in this case it's a dummy parameter. In any case, page-chooser.ts just receives the message AddSelectedPage and knows to add the currently selected page.